prov/efa/src/rxr/rxr_pkt_type_base.c

```diff
@@ -136,6 +140,14 @@ int rxr_pkt_init_data_from_tx_entry(struct rxr_ep *ep,
 	assert(tx_iov_index < tx_entry->iov_count);
 	assert(tx_iov_offset < tx_entry->iov[tx_iov_index].iov_len);
 
+	ret = efa_ep_use_p2p(efa_ep, desc);
```

Comment:
Can you elaborate on this change?

On the TX side, if the device type is initialized, the EFA device will be able to send from the hmem memory buffer? so I am not sure why copy is necessary?

Reply:
We'll need to copy if the user sets the p2p option to DISABLED regardless of whether the EFA device supports it or not. There may be a scenario where that device to device transfer will fail that Libfabric doesn't know about, but the user does.

Reply:
I see. Thanks!